examples/examples/check_extrinsic_events.rs

```diff
@@ -0,0 +1,142 @@
+/*
+	Copyright 2019 Supercomputing Systems AG
+	Licensed under the Apache License, Version 2.0 (the "License");
+	you may not use this file except in compliance with the License.
+	You may obtain a copy of the License at
+
+		http://www.apache.org/licenses/LICENSE-2.0
+
+	Unless required by applicable law or agreed to in writing, software
+	distributed under the License is distributed on an "AS IS" BASIS,
+	WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+	See the License for the specific language governing permissions and
+	limitations under the License.
+*/
+
+use sp_keyring::AccountKeyring;
+use substrate_api_client::{
+	ac_node_api::EventDetails,
+	ac_primitives::{AssetRuntimeConfig, Config, ExtrinsicSigner},
+	extrinsic::BalancesExtrinsics,
+	rpc::JsonrpseeClient,
+	Api, GetAccountInformation, SubmitAndWatch, TransactionStatus, XtStatus,
+};
+
+// To test this example with CI we run it against the Substrate kitchensink node, which uses the asset pallet.
+// Therefore, we need to use the `AssetRuntimeConfig` in this example.
+// ! However, most Substrate runtimes do not use the asset pallet at all. So if you run an example against your own node
+// you most likely should use `DefaultRuntimeConfig` instead.
+
+type Hash = <AssetRuntimeConfig as Config>::Hash;
+
+#[tokio::main]
+async fn main() {
+	env_logger::init();
+
+	// Initialize api and set the signer (sender) that is used to sign the extrinsics.
+	let alice_signer = AccountKeyring::Alice.pair();
+	let client = JsonrpseeClient::with_default_url().unwrap();
+	let mut api = Api::<AssetRuntimeConfig, _>::new(client).unwrap();
+	api.set_signer(ExtrinsicSigner::<AssetRuntimeConfig>::new(alice_signer));
+
+	let alice = AccountKeyring::Alice.to_account_id();
+	let balance_of_alice = api.get_account_data(&alice).unwrap().unwrap().free;
+	println!("[+] Alice's Free Balance is {balance_of_alice}\n");
+
+	let bob = AccountKeyring::Bob.to_account_id();
+	let balance_of_bob = api.get_account_data(&bob).unwrap().unwrap_or_default().free;
+	println!("[+] Bob's Free Balance is {balance_of_bob}\n");
+
+	// First we want to see the events of a failed extrinsic.
```

Comment:
How do you want to do it?

Reply:
Good Input - let me do something about that

Reply:
https://github.com/scs/substrate-api-client/issues/617

I think that should be done in a different PR, as it affects the API. What do you think? 


Regarding your point of
> submit_and_watch_extrinsic_until_without_event doesn't thrown an error when the extrinsic has failed onchain.
What do you propose for this? I think the comments are quite telling already.. so I removed the `without_events` from the examples. It should not be used if one does not know what one is doing.

Reply:
https://github.com/scs/substrate-api-client/issues/617 makes sense. Adding the report to the Error gives access to all the information we have collected.

Reply:
> submit_and_watch_extrinsic_until_without_event doesn't thrown an error when the extrinsic has failed onchain.
> What do you propose for this? I think the comments are quite telling already.. so I removed the `without_events` from the examples. It should not be used if one does not know what one is doing.

Not showing any examples is just a way of forcing the user not to use it. Is that really what we want? 
I'm not sure the comments are clear enough. Because we're talking about an error when an extrinsic has failed on the chain, and not an event, the user may think that `submit_and_watch_extrinsic_until_without_event` informs her when an extrinsic fails on the chain. Without studying the code, they may miss this point. 
Maybe add an example `check_extrinsic_without_events` similar to `check_extrinsic_events`? We could do it in another issue. Maybe I'm too picky, so it's up to you to decide.

Reply:
Created a  follow up issue in https://github.com/scs/substrate-api-client/issues/627